mutex_map.go

```diff
@@ -2,53 +2,43 @@ package nex
 
 import "sync"
 
+// RawMap implements a map type with helper functions to operate it.
+type RawMap[K comparable, V any] struct {
+	real map[K]V
+}
```

Comment:
This can just be made into a type alias rather than creating a whole struct

```suggestion
// RawMap implements a map type with helper functions to operate it.
type RawMap[K comparable, V any] map[K]V
```

Then it can just be used like a regular map, how `real` is being used internally. Though I'm not sure what the benefit of this type is right now, since all it seems to do is wrap a map and then give it helper functions for things Go already supports?

Like why use `RawMap.Delete()` rather than just using a normal `map` and the built in `delete()`?